pkgs/development/go-modules/generic/default.nix

```diff
@@ -20,6 +20,8 @@
 , vendorSha256 ? null
 # Whether to delete the vendor folder supplied with the source.
 , deleteVendor ? false
+# Whether to use the vend tool to regenerate the vendor directory.
+, useVend ? false
```

Comment:
Would create any harm to run this by default beside potential bugs in the tool?

Reply:
mhm. It might break our checksums again... For that reason we might want to test it.

Reply:
`useVend` is not very descriptive. Could be rename it to something like vendor C code?

Reply:
A few arguments against running this by default.
1) This is a lot more cpu expensive, it's longer to run and downloads everything twice.
2) I don't expect the upstream support will be very good compared to go. If go updates to a new version, I don't want that version updated blocked for everything by vend support.
3) This is a super weird thing to do. Almost no one else uses this program, and it's super fringe. I don't want nix users encountering too many differences from the normal go development tooling unless we actually need it. I know if I encountered bugs caused by this program when packaging a bog standard well behaved no c code go program that works with the standard tooling, I'd be mightily upset.

Reply:
useVend -> runVend. The key is to make sure that reading the option conveys we're running a pgroam to a user. vendorFull makes this sound like some supported thing, rather than a weird program that helps in a couple of corner cases.